### PR TITLE
Support creating keychain from multiple volume secrets

### PR DESCRIPTION
**Pull request by tylerphelan**
- continue supporting historical env var used to set the path to the secret dir

### CODE REVIEW

File: pkg/dockercreds/volume_secrets_keychain.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+const (
+	SecretFilePathEnv  = "CREDENTIAL_PROVIDER_SECRET_PATH" // historically only support one path
+	SecretFilePathsEnv = "CREDENTIAL_PROVIDER_SECRET_PATHS"
```
Comment (dumez-k) [inline pkg/dockercreds/volume_secrets_keychain.go:12]: should we document this new env variable somewhere as well?

**Review by tomkennedy513 — APPROVED** [review]

File: pkg/dockercreds/volume_secrets_keychain.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		if err != nil {
+			return nil, err
+		}
+		secrets.Append(newSecrets)
+	}
+
+	return secrets, nil
+}
+
+func getSecretFolders() []string {
```
Comment (matthewmcnew) [inline pkg/dockercreds/volume_secrets_keychain.go:29]: Could we use the `directory` term over `folders` to be consisent?

File: pkg/dockercreds/volume_secrets_keychain.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	secretFoldersEnvVal, ok := os.LookupEnv(SecretFilePathsEnv)
+	if !ok {
+		secretFolderEnvVal, ok := os.LookupEnv(SecretFilePathEnv)
+		if !ok {
+			return []string{}
+		}
+		return []string{secretFolderEnvVal}
+	}
+
+	return strings.Split(secretFoldersEnvVal, ",")
```
Comment (jromero) [inline pkg/dockercreds/volume_secrets_keychain.go:39]: Wouldn't it be more expected to split paths by colon (`:`) as opposed to commas? (prior art: [PATH](https://en.wikipedia.org/wiki/PATH_(variable)))

**Comment by matthewmcnew** [comment]
I would like to propose that we merge https://github.com/pivotal/kpack/pull/837 instead of growing the scope of credentials that get mounted into the kpack controller. With https://github.com/pivotal/kpack/pull/837 and the new v1alpha2 api all clusterstores/clusterstacks/lifecycle can configure a service account with image pull secrets directly in the api.

**Pull request closed without merging**